docs/naps/9-multiple-canvases.md

```diff
@@ -43,210 +49,291 @@ Several plugins and examples have been created to address these limitations, for
 
-### Architecture
-napari architecture is based on the MVC pattern. The model layer comprises a `ViewerModel` and a list of Layer models (subclasses of a base `Layer`). There are seven layer types, each with a corresponding view type. Currently models and views are paired 1:1, and the correlation is stored in a map (`layer_to_visual`) on the `VispyCanvas`. Figure 1 shows the class relationships for the base model types and the Image layer types (for brevity - other layer types have similar connectivity).
+## Implementation
 
-```{figure} _static/multicanvas-napari-architecture-today.png
-:name: nap-9-fig-1
+(part-1-view-model)=
+### Part 1: View model
+
+* Introduce a minimally disruptive `_views` attribute on the ViewerModel, and implement a `View` model to hold a `Layerlist`, a `Dims`, and a `Camera` (as well as some related concepts and models).
+* using a `SelectableEventedList` for `_views` lets us easily introduce the concept of an `active` View. This, together with some simple dispatch of current viewer elements to the active view, will make this part of the implementation a drop-in replacement.
+* At this stage, only the active view will be displayed. This will leave the GUI identical to before. When a new view becomes active, the contents of the QtViewer and vispy canvas are simply swapped to be linked with the new layerlist, camera, and dims.
+* Similarly, existing events and event callbacks will connected to the active view only.
+* Add APIs to add, remove and select views
+
+The concept of an "active" view will work in service of minimizing API changes. This will allow existing APIs on the main Viewer/ViewerModel to remain and simply delegate to the active view.
+
+Additionally, by providing access to the components of the active view through the `ViewerModel`, we can also seamlessly transition to a multiview system without breaking any high level API.
+
+Currently, the `ViewerModel` (when stripped down to its barebones) contains the following fields and functionality:
+
+```py
+class ViewerModel:
+    camera: Camera
+    cursor: Cursor
+    dims: Dims
+    grid: GridCanvas
+    layers: LayerList
+    help: str
+    status: Union[str, dict]
+    tooltip: Tooltip
+    theme: str
+    title: str
+    _overlays: EventedDict[str, Overlay]
+    _layer_slicer: _LayerSlicer
+```
 
-Fig. 1: napari architecture today.
+With this NAP and the above considerations, we expect to separate the above components roughly as follows:
+
+```py
+class View:
+    camera: Camera
+    cursor: Cursor
+    dims: Dims
+    grid: GridCanvas
```

Comment:
I find this weird, because I thought that GridCanvas was the first implementation of multi(canvas/view). That is, the way I envisioned things instead is that grid remains in the ViewerModel and triggers the creation of many `View`s with linked `Camera`s. What am I missing?